src/preprocessor.ts

```diff
@@ -55,14 +55,27 @@ export function process(
     fileName: filePath,
   });
 
+  let tsTranspiledText = tsTranspiled.outputText;
+  if (tsJestConfig.ignoreCoverageForAllDecorators === true) {
+    tsTranspiledText = tsTranspiledText.replace(
+      /__decorate/g,
+      '/* istanbul ignore next */__decorate',
+    );
+  } else {
+    tsTranspiledText = tsTranspiledText.replace(
+      /(__decorate\(\[\r?\n[^\n\r]*)\/\*\s*istanbul\s*ignore\s*decorate(.*)\*\//g,
+      '/* istanbul ignore next$2*/$1',
+    );
+  }
+
```

Comment:
Could you tell me what this else block is doing?

Ideally, this shouldn't change any behavior for folks who haven't set this flag

Reply:
So I made two options to ignore decorators. One is with setting the config param ignoreCoverageForAllDecorators. Thi swill ignore all decorators and it's useful for Angular where you don't have control over the code. This is described in #454. But the other option is to add the /* istanbul ignore decorator \*/ and it will ignore only this decorator. I prefer the second option because I like to have control over which part of code I want to ignore in coverage. Both are optional and if you don't use them it will not modify anything.

Reply:
What I meant was that if `tsJestConfig.ignoreCoverageForAllDecorators` is not set, the source code should not be modified for decorator coverage at all. In this case, it looks like it is being modified which will affect even users who haven't set this flag.

Reply:
I can add another config for that. There are basically 3 scenarios. First is for users who don't want to change anything, second is for someone like me who want's to control which decorators are ignored and the third is for users who want to ignore all decorators. The last one is enabled by ignoreCoverageForAllDecorators config param. The first and second I didn't make any config setting but you control it by providing (or not) the /* istanbul ignore decorator \*/ comment. But I can add another config param for that if you believe it is necessary.

Reply:
Yes, I think that would be the way to go to avoid any side effects for users who aren't setting the flag at all.

I'll take a look at this again after work and merge it if it's ready (flags, docs and updated version patch). Thanks!

Reply:
Done. Updated the readme too.